container_pipeline/utils.py

```diff
@@ -211,3 +211,30 @@ def complete(self):
             os.remove(self._path)
             self.logger.info('Removing lock file for {} at {}'.format(
                 self.name, self._path))
+
+
+def form_targetfile_link(git_URL, git_path, git_branch, target_file):
+    """
+    Helper function to generate Dockerfile link.
+    :param git_URL: The url of the git repository.
+    :param git_path: The path, relative to the git repository root,
+                     where the file resides.
+    :param git_branch: The repository branch where the file resides.
+    :param target_file: The name of the target file.
+    :return: The reachable link to the dockerfile.
+    """
+
+    # TODO : Move this into a lib.
+
+    link_url = None
+    if "github" in git_URL or "gitlab" in git_URL:
+        link_url = str.format(
+            "{git_url}/blob/{git_branch}/{git_path}/{target_file}",
+            git_url=git_URL,
+            git_branch=git_branch,
+            git_path=git_path if not git_path.startswith("/") else "/" +
+                                                                   git_path,
+            target_file=target_file
+        )
+    return link_url
+
```

Comment:
The pep8 error read now new line at the end of file, while we are adding an extra line here, causing CI failure at pep8 gate again.

You can run `pep8 utils.py` before doing git push.
Running `autopep8 -i utils.py` should fix utils.py inline.